code/gateway/dataaccessOverwrite.q

```diff
@@ -0,0 +1,71 @@
+\d .dataaccess
+
+// Dynamic routing finds all processes with relevant data
+attributesrouting:{[options;procdict]
+    // Get the tablename and timespan
+    timespan:`long$options[`starttime`endtime];
+    //adjust rdb partition range to account for period end time)
+    if[`rdb in key procdict;procdict[`rdb]:(first[procdict `rdb];-1+`long$01D00 + last procdict `rdb)];
+    // See if any of the provided partitions are with the requested ones
+    procdict:{[x;timespan] (all x within timespan) or any timespan within x}[;timespan] each procdict;
+    // Only return appropriate dates
+    types:(key procdict) where value procdict;
+    // If the dates are out of scope of processes then error
+    if[0=count types;
+        '`$"gateway error - no info found for that table name and time range. Either table does not exist; attributes are incorect in .gw.servers on gateway, or the date range is outside the ones present"
+       ];
+    :types;
+    };
+
+// function to adjust the queries being sent to processes to prevent overlap of
+// time clause and data being queried on more than one process
+adjustqueries:{[options;part]
+    // if only one process then no need to adjust
+    if[2>count p:options`procs;:options];
+    // get the partitions that are required by each process
+    tabname:options[`tablename];
+    // Remove duplicate servertypes from the gw.servers
+    servers:select from .gw.servers where i=(first;i)fby servertype;
```

Comment:
why are we doing this? will this break our load balancing?

Reply:
to prevent the partitions of hdb2 being including in possible partitions. borrowed this existing logic from the partdict function so shouldn't break anything